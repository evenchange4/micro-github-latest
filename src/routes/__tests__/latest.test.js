/* eslint global-require: 0, no-throw-literal: 0 */

beforeEach(() => {
  jest.resetModules();
  jest.mock('micro', () => ({
    send: (res, code, content) => ({ res, code, content }),
  }));
  jest.mock('../../utils/redirect', () => (res, statusCode, url) => ({
    res,
    statusCode,
    url,
  }));
});

it('should return latest without name', async () => {
  jest.doMock('../../utils/API', () => ({
    getLatestRelease: (owner, repo) => ({
      owner,
      repo,
    }),
  }));
  jest.doMock('../../utils/getAssets', () => () => ['1', '2', '3']);

  const latest = require('../latest');

  expect(
    await latest({ params: { owner: 'owner', repo: 'repo' } }),
  ).toMatchSnapshot();
});

it('should return latest with name', async () => {
  jest.doMock('../../utils/API', () => ({
    getLatestRelease: (owner, repo) => ({
      owner,
      repo,
    }),
  }));
  jest.doMock('../../utils/getAssets', () => () => ['1', '2', '3']);

  const latest = require('../latest');

  expect(
    await latest({ params: { owner: 'owner', repo: 'repo', name: '2' } }),
  ).toMatchSnapshot();
});

it('should return latest with 404', async () => {
  jest.doMock('../../utils/API', () => ({
    getLatestRelease: (owner, repo) => ({
      owner,
      repo,
    }),
  }));
  jest.doMock('../../utils/getAssets', () => () => []);

  const latest = require('../latest');

  expect(
    await latest({ params: { owner: 'owner', repo: 'repo', name: '2' } }),
  ).toMatchSnapshot();
});

it('should return latest with error', async () => {
  jest.doMock('../../utils/API', () => ({
    getLatestRelease: () => {
      throw new Error('error');
    },
  }));
  jest.doMock('../../utils/getAssets', () => () => []);

  const latest = require('../latest');

  expect(
    await latest({ params: { owner: 'owner', repo: 'repo' } }),
  ).toMatchSnapshot();
});
