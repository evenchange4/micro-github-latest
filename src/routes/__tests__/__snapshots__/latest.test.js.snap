// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return latest with 404 1`] = `
Object {
  "code": 404,
  "content": [Error: Asset not found.],
  "res": undefined,
}
`;

exports[`should return latest with error 1`] = `
Object {
  "code": 404,
  "content": [Error: error],
  "res": undefined,
}
`;

exports[`should return latest with name 1`] = `
Object {
  "res": undefined,
  "statusCode": 302,
  "url": "2",
}
`;

exports[`should return latest without name 1`] = `
Object {
  "res": undefined,
  "statusCode": 302,
  "url": "1",
}
`;
